solutions/usaco-1230.mdx

```diff
@@ -2,10 +2,19 @@
 id: usaco-1230
 source: USACO Silver 2022 Open
 title: Visits
-author: Aditya Gupta
+author: Aditya Gupta, Chuyang Wang
 ---
 
-[Official Analysis](http://www.usaco.org/current/data/sol_prob1_silver_open22.html)
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_prob1_silver_open22.html)
+
+
+## Explanation
+
+As each cow $i$ only wants to visit one of the other cows $j$, we can interpret the input as a functional graph with edges $i \to j$. The resulting graph is not necessarily connected, so we have to find out all connected components in this graph and solve them separately. 
+
+For each connected component of the functional graph, let's observe that there must be one and only one cycle in it. Moreover, all cows can visit their buddies before their departure except one who wants to visit the cow initiating the visit chain in the cycle. For other paths that lead to the cycle, we can always visit them first before processing the cycle, so they will always be able to "moo". Therefore, the maximum amount of "moos" is the sum of all $v$ in the connected component subtracted by the minimum $v_i$ in the cycle. 
```

Comment:
add something like
we subtract the min $v_i$ in the cycle because it represents the initiating cow who won't be able to moo